gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2324,15 +2514,31 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
         if( poFieldDefn->GetWidth() == 1 )
         {
             nType = NC_CHAR;
-            status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1,
+
+            if(!m_bLegacyCreateMode)
+            {
+                nVarID = m_poDS->vcdf.nc_def_vvar(pszVarName, nType, 1, &nMainDimId);
```

Comment:
why nc_def_vvar and not nc_def_var ?

Reply:
Resizable dimensions-

I think this relates what we just talked about. So I'll resolve this once the paths are combined.